iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/HiveIcebergRESTCatalogClientAdapter.java

```diff
@@ -0,0 +1,2463 @@
+  public static final String CATALOG_NAME = "iceberg.catalog";
+  public static final String DB_OWNER = "owner";
+  public static final String DB_OWNER_TYPE = "ownerType";
+  public static final String SERDE_INFO = "serdeInfo";
+  public static final String DEFAULT_INPUT_FORMAT_CLASS = "org.apache.iceberg.mr.hive.HiveIcebergInputFormat";
+  public static final String DEFAULT_OUTPUT_FORMAT_CLASS
+      = "org.apache.iceberg.mr.hive.HiveIcebergOutputFormat";
+  public static final String DEFAULT_SERDE_CLASS = "org.apache.iceberg.mr.hive.HiveIcebergSerDe";
+  public static final String CATALOG_CONFIG_PREFIX = "iceberg.catalog.";
+  public static final Set<String> PROPERTIES_TO_REMOVE = ImmutableSet
+      // We don't want to push down the metadata location props to Iceberg from HMS,
+      // since the snapshot pointer in HMS would always be one step ahead
+      .of(BaseMetastoreTableOperations.METADATA_LOCATION_PROP,
+          BaseMetastoreTableOperations.PREVIOUS_METADATA_LOCATION_PROP);
+  private final Configuration conf;
+  private RESTCatalog restCatalog;
+  private final HiveMetaHookLoader hookLoader;
+
+  private final long maxHiveTablePropertySize;
+
+  public HiveIcebergRESTCatalogClientAdapter(Configuration conf, HiveMetaHookLoader hookLoader) {
+    this.conf = conf;
+    this.hookLoader = hookLoader;
+    this.maxHiveTablePropertySize = conf.getLong(HiveOperationsBase.HIVE_TABLE_PROPERTY_MAX_SIZE,
+          HiveOperationsBase.HIVE_TABLE_PROPERTY_MAX_SIZE_DEFAULT);
+  }
+
+  @Override
+  public boolean isCompatibleWith(Configuration configuration) {
+    return false;
+  }
+
+  @Override
+  public void setHiveAddedJars(String addedJars) {
+
+  }
+
+  @Override
+  public boolean isLocalMetaStore() {
+    return false;
+  }
+
+  @Override
+  public void reconnect() throws MetaException {
+    SessionCatalog.SessionContext context = SessionCatalog.SessionContext.createEmpty();
+    String catalogName = conf.get(CATALOG_NAME);
+    Map<String, String> properties = getCatalogProperties(conf, catalogName);
+    restCatalog = (RESTCatalog) CatalogUtil.buildIcebergCatalog(catalogName, properties, conf);
+    restCatalog.initialize(catalogName, properties);
+  }
+  private static Map<String, String> getCatalogProperties(
+      Configuration conf, String catalogName) {
+    Map<String, String> catalogProperties = Maps.newHashMap();
+    String keyPrefix = CATALOG_CONFIG_PREFIX + catalogName;
+    conf.forEach(config -> {
+      if (config.getKey().startsWith(keyPrefix)) {
+        catalogProperties.put(
+            config.getKey().substring(keyPrefix.length() + 1),
+            config.getValue());
+      }
+    });
+    return catalogProperties;
+  }
+  @Override
+  public void close() {
+
+  }
+
+  @Override
+  public void setMetaConf(String key, String value) throws MetaException, TException {
+
+  }
+
+  @Override
+  public String getMetaConf(String key) throws MetaException, TException {
+    return "";
```

Comment:
Why doesn't this method returns the value from conf?

Reply:
Because there is no matching alternative in Iceberg REST catalog API.